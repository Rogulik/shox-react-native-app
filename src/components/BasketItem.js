import React from "react";
import {
  Image,
  Platform,
  StyleSheet,
  Text,
  TouchableNativeFeedback,
  TouchableOpacity,
  View,
} from "react-native";
import colors from "../constants/Colors";
import Button from "./Button";

const BasketItem = (props) => {
  const { title, price, image, onViewDetail, onRemoveFromBasket } = props;

  let TouchableCmp =
    Platform.OS === "android" && Platform.Version >= 21
      ? TouchableNativeFeedback
      : TouchableOpacity;

  return (
    <View style={styles.product}>
      <View style={styles.touchable}>
        <TouchableCmp onPress={onViewDetail} useForeground>
          <View style={styles.mainContainer}>
            <View style={styles.imageContainer}>
              <Image style={styles.image} source={{ uri: image }} />
            </View>
            <View style={styles.content}>
              <Text style={styles.title} testID="title">
                {title}
              </Text>
              <Text testID="price" style={styles.price}>
                ${price.toFixed(2)}
              </Text>
            </View>

            <View style={styles.buttonsContainer}>
              <Button
                color={colors.primary}
                title="View Details"
                onPress={onViewDetail}
              />
              <Button
                color={colors.primary}
                title="Remove from Cart"
                onPress={onRemoveFromBasket}
              />
            </View>
          </View>
        </TouchableCmp>
      </View>
    </View>
  );
};

export default BasketItem;

const styles = StyleSheet.create({
  product: {
    shadowColor: "black",
    shadowOpacity: 0.26,
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 8,
    elevation: 5,
    borderRadius: 10,
    backgroundColor: "white",
    height: 80,
    width: "90%",
    margin: 20,
  },
  touchable: {
    overflow: "hidden",
    borderRadius: 10,
  },
  image: {
    width: "100%",
    height: "100%",
  },
  title: {
    fontSize: 14,
    textAlign: "center",
    fontFamily: "open-sans-bold",
  },
  price: {
    fontSize: 14,
    color: "#888",
    fontFamily: "open-sans",
  },
  buttonsContainer: {
    flexDirection: "row",
    justifyContent: "space-around",
    alignItems: "center",
    width: "40%",
  },
  content: {
    justifyContent: "space-around",
    alignItems: "center",
    width: "40%",
  },
  imageContainer: {
    height: "100%",
    width: "20%",
    borderTopLeftRadius: 10,
    borderTopRightRadius: 10,
    overflow: "hidden",
  },
  mainContainer: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-around",
    width: "100%",
    height: "100%",
  },
});
