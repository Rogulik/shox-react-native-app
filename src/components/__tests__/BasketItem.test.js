import React from "react";
import { create } from "react-test-renderer";
import BasketItem from "../BasketItem";

describe("BasketItem", () => {
  it("should render Basket item", () => {
    const tree = create(<BasketItem price={2} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  //   it("should render price equal of 8 with fixed two decimal points", () => {
  //     const tree = create(<BasketItem price={8} />).root;

  //     expect(tree.findByProps({ testID: "price" }).children).toEqual(["8.00"]);
  //   });

  it("should render title product equal test", () => {
    const tree = create(<BasketItem title="test" price={8} image="" />).root;

    console.log(tree.findByProps({ testID: "title" }).children);

    // expect(tree.findByProps({ testID: "title" }).children).toEqual("test");
  });
});
