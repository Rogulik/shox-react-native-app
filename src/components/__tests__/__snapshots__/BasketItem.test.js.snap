// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasketItem should render Basket item 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 10,
      "elevation": 5,
      "height": 80,
      "margin": 20,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.26,
      "shadowRadius": 8,
      "width": "90%",
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 10,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": "100%",
            "justifyContent": "space-around",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "borderTopLeftRadius": 10,
              "borderTopRightRadius": 10,
              "height": "100%",
              "overflow": "hidden",
              "width": "20%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": undefined,
              }
            }
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "space-around",
              "width": "40%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "open-sans-bold",
                "fontSize": 14,
                "textAlign": "center",
              }
            }
            testID="title"
          />
          <Text
            style={
              Object {
                "color": "#888",
                "fontFamily": "open-sans",
                "fontSize": 14,
              }
            }
            testID="price"
          >
            $
            2.00
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-around",
              "width": "40%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#C2185b",
                "borderRadius": 4,
                "elevation": 3,
                "height": 40,
                "justifyContent": "center",
                "paddingHorizontal": 2,
                "paddingVertical": 8,
                "width": 60,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 10,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              View Details
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#C2185b",
                "borderRadius": 4,
                "elevation": 3,
                "height": 40,
                "justifyContent": "center",
                "paddingHorizontal": 2,
                "paddingVertical": 8,
                "width": 60,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 10,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Remove from Cart
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
